src/components/visualizations/chart/chartOptionsBuilder.ts

```diff
@@ -1781,8 +1811,12 @@ export function getChartOptions(
         };
     }
 
-    const tooltipFn = isTreemap(type) ? generateTooltipTreemapFn(viewByAttribute, stackByAttribute, config) :
-        generateTooltipFn(viewByAttribute, type, config);
+    const tooltipFn =
+        isTreemap(type) ?
+            generateTooltipTreemapFn(viewByAttribute, stackByAttribute, config) :
+            isViewByTwoAttributes ?
+                generateTooltipTwoAttributesFn(viewByAttribute, viewByParentAttribute, config) :
+                generateTooltipFn(viewByAttribute, type, config);
```

Comment:
nested ternary condition should be separated to new var

Reply:
changed to if else